app/models/user/permissions.rb

```diff
@@ -79,9 +79,12 @@ def existing_service_ids
   # but then it is much harder to test MemberPermission#service_ids=
   # returns [] if no services are enabled, and nil if all (current and future) services are enabled
   def member_permission_service_ids
-    return nil if admin? || !services_member_permission
-    permitted_service_ids = services_member_permission.try(:service_ids) || []
-    permitted_service_ids & existing_service_ids
+    if admin? || !services_member_permission
```

Comment:
This change was for memoization, but I found out that it gives unexpected results when the permissions are changed. So I guess we can't really memoize here, because we want the values to be updated correctly.